streamer/src/nonblocking/quic.rs

```diff
@@ -85,6 +89,120 @@ struct PacketAccumulator {
     pub chunks: Vec<PacketChunk>,
 }
 
+struct StakedStreamLoadEMA {
+    current_load_ema: AtomicU64,
+    load_in_recent_interval: AtomicU64,
+    last_update: RwLock<Instant>,
+    stats: Arc<StreamStats>,
+}
+
+impl StakedStreamLoadEMA {
+    fn new(stats: Arc<StreamStats>) -> Self {
+        Self {
+            current_load_ema: AtomicU64::default(),
+            load_in_recent_interval: AtomicU64::default(),
+            last_update: RwLock::new(Instant::now()),
+            stats,
+        }
+    }
+
+    fn ema_function(current_ema: u128, recent_load: u128) -> u128 {
```

Comment:
do we anticipate results actually being large enough to require `u128`?  my intuition is that we'll only need `u128` to avoid overflowing during arithmetic and that we can otherwise use `u64` everywhere

Reply:
Results will not be large enough for a `u128`. We can downscope it, but it'll require downscoping at a few places.  The only place we actually use the result is here.

```
                    if stream_counter.stream_count.load(Ordering::Relaxed) as u128
                        >= max_streams_per_throttling_interval
```

I am split minded to whether to do the downscope, or just use the current comparison code where it upscopes the counter. Technically, upscoping u64 to u128 is safer.

Reply:
i think sticking to `u64` where we can, the upcasting for the arithmetic is probably preferable. `u64` being native word size on the target arch and all

Reply:
I'll push an update. Do you have any other feedback that I can address along with it?